js/CADWorker/CascadeStudioWorkerState.js

```diff
@@ -0,0 +1,26 @@
+export const messageHandlers = {};
+
+export let argCache = {};
+export let currentLineNumber = 0;
+export let currentOp = "";
+export let currentShape;
+export let externalShapes = {};
+export let GUIState = {};
+export let oc = null;
+export let opNumber = 0; // This keeps track of the progress of the evaluation
+export let usedHashes = {};
+
+export const setArgCache = val => (argCache = val);
+export const setCurrentOp = val => (currentOp = val);
+export const setCurrentShape = val => (currentShape = val);
+export const setCurrentLineNumber = val => (currentLineNumber = val);
+export const resetExternalShapes = () => (externalShapes = {});
+export const setGUIState = val => (GUIState = val);
+export const setOc = ocInit => (oc = ocInit);
+export const setOpNumber = val => (opNumber = val);
+export const setUsedHashes = val => (usedHashes = val);
+
+
+// I can't see anywhere, where the following globals are used, variables with these names exist, but they are scoped.
+let fullShapeEdgeHashes = {};
+let fullShapeFaceHashes = {};
```

Comment:
I've left these here just to remind myself to ask about them, I think they can be removed. it's just this line made me think I need to be careful https://github.com/zalo/CascadeStudio/pull/58/files#diff-7d888ae3a42f72ffc3319a8a7a9476c873382cdfa729f59ef60b2c5ea257d44cL3

Reply:
I believe they're only used here:
https://github.com/zalo/CascadeStudio/blob/master/js/CADWorker/CascadeStudioShapeToMesh.js

Must be vestigial from when I was experimenting with different instanced-mesh, mouse-over/tooltip indexing techniques...